### DISCUSSION TITLE
Export data

### DISCUSSION THREAD

**Discussion by paur94** [KevinVandy/material-react-table#471] [Q&A] [answered]
Hello everyone!
Can I export only filtered data? Have you an example of that? Thank you very much.

**Comment by KevinVandy** [comment, marked as answer]
The official [Export example](https://www.material-react-table.com/docs/examples/data-export) already shows you how to do this

**Comment by DeekshithBanjan** [comment]
Hi, I am able to export all rows and selected rows through this function.

`const handleExportData = (table) => {
    console.log("table", table, table.getSelectedRowModel().rows);
    if (table.getSelectedRowModel().rows.length) {
      csvExporter.generateCsv(
        table.getSelectedRowModel().rows.map((row) => row.original)
      );
    } else {
      csvExporter.generateCsv(data);
    }
  };`

But is it possible to download selected columns data, filtered data and grouped data to csv?


![image](https://user-images.githubusercontent.com/127406455/235840401-44ca1b3e-d692-4bdd-99fd-9163345e8950.png)
![image](https://user-images.githubusercontent.com/127406455/235861968-cb19736d-944a-4174-b1e1-f792ae2b9873.png)

**Reply by KevinVandy** [reply]
yeah, just filter the `row.original` values to only have the columns you want. Or use `row.getVisibleCells` instead
